elastic/kibana
# [Infrastructure UI] Inform about missing data view while setting index pattern

Closes #152885 
## Summary
This PR adds information about missing metrics indices/remote cluster connection on the settings page.

## Bugfixes

While testing I saw that on metrics explorer page we don't show no cluster connection page when there is no connection to the cluster and added it in this commit: [a1a20f2](https://github.com/elastic/kibana/pull/156592/commits/a1a20f2767c5a4aefeeaaeace87c860f0c24a535)

<img width="2543" alt="image" src="https://user-images.githubusercontent.com/14139027/236174332-8df06202-fed4-4a0a-b732-782391ebfaab.png">


## Testing

- Connect to a remote cluster
- Open the settings page
- Inside metrics indices input set a not existing index and click on apply
- A warning callout should appear to inform you that the indices are missing
- Navigate to the inventory (or metric explorer) page
- No data prompt should be displayed 

https://user-images.githubusercontent.com/14139027/236161644-2b96c76b-8380-43a1-a945-e17dc56ab079.mov


<img width="2548" alt="image" src="https://user-images.githubusercontent.com/14139027/236160110-1b23202c-4783-485a-b124-4234498c00a9.png">

<img width="2544" alt="image" src="https://user-images.githubusercontent.com/14139027/236175868-8e625e05-beb4-4624-b9b7-e6930ddbaa71.png">




- Simulate lost connection to a remote cluster ( comment out settings in kibana yml)
- Log in again
- Open the settings page with the previous metrics indices added
- An error callout should inform you that there is no connection to the remote cluster
- Navigate to the inventory (or metric explorer) page
- No remote cluster connection prompt should be displayed 


https://user-images.githubusercontent.com/14139027/236161071-6f8c4a6f-1bca-403f-86ce-733b2c8195e7.mov

<img width="2547" alt="image" src="https://user-images.githubusercontent.com/14139027/236177076-9af78622-6f43-46ee-90b9-ca2081bca4ce.png">

![image](https://user-images.githubusercontent.com/14139027/236176369-c154c965-5f44-41b4-a3fb-a39aca691090.png)
